src/r4300/new_dynarec/new_dynarec.c

```diff
@@ -10127,91 +10127,150 @@ int new_recompile_block(int addr)
   // to use, which can avoid a load-use penalty on certain CPUs.
   for(i=0;i<slen-1;i++)
   {
-    if(!i||(itype[i-1]!=UJUMP&&itype[i-1]!=CJUMP&&itype[i-1]!=SJUMP&&itype[i-1]!=RJUMP&&itype[i-1]!=FJUMP))
+    if(count_free_regs(regs[i].regmap)>minimum_free_regs[i])
```

Comment:
You're right the new_dynarec already is a bit hard to read, no need to make it worse ;-)
I updated the pull request.